### ISSUE TITLE
An annoying and somewhat game-breaking glitch.

### ISSUE THREAD

**Issue by Cookie-Kurokami** [CammiePone/Hookshot#15]
In my server, almost every other use or so, the gravity-negating aspect of the grappling hook (the feature that prevents the player from simply falling to the ground when attempting to go up or cross a gap) doesn't get cancelled upon releasing the hook, causing the player to continue floating in the air until they fire the hook again. I have no idea what is causing this, and since it's neither crashing the game nor producing errors in the console I don't exactly have anything to paste here to be looked at. On the off chance that this is a known issue, might a fix for it be coming soon?

**Comment by CammiePone** [comment]
uhhh okay so the only way I know that it's caused is when an entity the player is hooked to dies while they're hooked to it, or the block they're hooked to is broken.

**Comment by Cookie-Kurokami** [comment]
The blocks aren't being broken and I'm not hooking to any entities. I fire
the hookshot, I latch onto a block and fly toward it, and then I let go and
there's a 50/50 chance I'll get stuck in the air. I'm also, strangely
enough, not able to stay hooked to something for long before the grapple
automatically disconnects - this is without the observer upgrade. Is it
possible that the the fired hook is, much like an arrow, an entity itself?
If so, I think it's possible it might be getting killed by one of my other
mods.

**Comment by CammiePone** [comment]
Well yea it is an entity. There really isn't any other way to do it other than with an entity.

But it does sound like maybe something else is killing it, in which case idk if I can fix it. It's supposed to disable any changes to the player I make if the hook is killed, so idk why it's not. Are you on the latest version of Hookshot, r18?

**Comment by Cookie-Kurokami** [comment]
Yeah. I updated to the latest version as soon as it came out hoping it
might fix the problem. Needless to say, it's still persisting. Are there
any known mod incompatibilities?

**Comment by CammiePone** [comment]
Not that could cause that to my knowledge. Kinda weird ngl

**Comment by LumberWizard** [comment]
Could you send your mod list and a link to the server world?

**Comment by Cookie-Kurokami** [comment]
Modpack can be downloaded from here: https://www.dropbox.com/s/32ddymqlnawgxn1/The%20Full%20Package.zip?dl=1

The server IP: play.ungodly.xyz

**Comment by LumberWizard** [comment]
I meant the actual world files, so I could test it in a debug environment. Does it happen in single player in a fresh world in the same pack?

**Comment by Cookie-Kurokami** [comment]
It's only happened on the server. Never had this problem in singleplayer.

**Comment by Cookie-Kurokami** [comment]
Hey, so, I think I've figured out exactly what's going on. If the sounds
are any indication, the grapple's hook is being picked up like an item,
similar to an arrow stuck in a block. Is there a way to perhaps completely
block this from happening in a future update?

**Comment by CammiePone** [comment]
Okay so that's 100% another mod's fault, and idk if there is anything I could do about it. If I knew what mod, I _might_ be able to find a fix, but without that knowledge, there's not much I can do.

**Comment by Cookie-Kurokami** [comment]
I believe it might be Fabrication's auto-pickup feature. I'm not certain
though, I'll have to test it later.

**Comment by Cookie-Kurokami** [comment]
I just tested it. It's not Fabrication's instant pickup.

**Comment by CammiePone** [comment]
Hm... really not sure what mod would be causing it. Best advice I can give is to just remove half the mods, see if it's still happening, if so remove half of those, if not swap to the other half, and repeat until you find the mod(s) responsible.
